test/cpp/interop/rpc_behavior_lb_policy.cc

```diff
@@ -0,0 +1,196 @@
+
+ private:
+  class Picker : public SubchannelPicker {
+   public:
+    Picker(RefCountedPtr<SubchannelPicker> delegate_picker,
+           RefCountedPtr<RpcBehaviorLbPolicy> parent)
+        : delegate_picker_(std::move(delegate_picker)), parent_(parent) {}
+
+    PickResult Pick(PickArgs args) override {
+      args.initial_metadata->Add(kRpcBehaviorMetadataKey,
+                                 parent_->rpc_behavior_);
+      // Do pick.
+      return delegate_picker_->Pick(args);
+    }
+
+   private:
+    RefCountedPtr<SubchannelPicker> delegate_picker_;
+    RefCountedPtr<RpcBehaviorLbPolicy> parent_;
+  };
+
+  class Helper : public ChannelControlHelper {
+   public:
+    explicit Helper(RefCountedPtr<RpcBehaviorLbPolicy> parent)
+        : parent_(std::move(parent)) {}
+
+    RefCountedPtr<grpc_core::SubchannelInterface> CreateSubchannel(
+        grpc_core::ServerAddress address,
+        const grpc_core::ChannelArgs& args) override {
+      auto subchannel = parent_->channel_control_helper()->CreateSubchannel(
+          std::move(address), args);
+      return subchannel;
+    }
+
+    void UpdateState(grpc_connectivity_state state, const absl::Status& status,
+                     RefCountedPtr<SubchannelPicker> picker) override {
+      parent_->channel_control_helper()->UpdateState(
+          state, status,
+          grpc_core::MakeRefCounted<Picker>(std::move(picker), parent_));
+    }
+
+    void RequestReresolution() override {
+      parent_->channel_control_helper()->RequestReresolution();
+    }
+
+    absl::string_view GetAuthority() override {
+      return parent_->channel_control_helper()->GetAuthority();
+    }
+
+    grpc_event_engine::experimental::EventEngine* GetEventEngine() override {
+      return parent_->channel_control_helper()->GetEventEngine();
+    }
+
+    void AddTraceEvent(TraceSeverity severity,
+                       absl::string_view message) override {
+      parent_->channel_control_helper()->AddTraceEvent(severity, message);
+    }
+
+   private:
+    RefCountedPtr<RpcBehaviorLbPolicy> parent_;
+  };
+
+  void ShutdownLocked() override {
+    grpc_pollset_set_del_pollset_set(delegate_->interested_parties(),
+                                     interested_parties());
+    delegate_.reset();
+  }
+
+  OrphanablePtr<LoadBalancingPolicy> delegate_;
+  std::string rpc_behavior_;
+};
+
+class RpcBehaviorLbPolicyFactory
+    : public grpc_core::LoadBalancingPolicyFactory {
+ private:
+  absl::string_view name() const override { return kRpcBehaviorLbPolicyName; }
+
+  grpc_core::OrphanablePtr<LoadBalancingPolicy> CreateLoadBalancingPolicy(
+      LoadBalancingPolicy::Args args) const override {
+    return grpc_core::MakeOrphanable<RpcBehaviorLbPolicy>(std::move(args));
+  }
+
+  absl::StatusOr<RefCountedPtr<LoadBalancingPolicy::Config>>
+  ParseLoadBalancingConfig(const Json& json) const override {
+    if (json.type() == Json::Type::kNull) {
```

Comment:
You can remove this block.  This case can't happen anymore, and I've been meaning to remove this check from all of our existing policies.

Reply:
Done